
The file br2.dat contains data on 1,080 houses sold in Baton Rouge, Louisiana, during mid-2005. We will be concerned with the selling price (PRICE), the size of the house in square feet (SQFT), and the age of the house in years (AGE). Define a new variable that measures house size in terms of hundreds of square feet, $SQFT100= SQFT/100$.

a. Estimate the following equation and report the results:
$$ \ln(PRICE) = \alpha_1 + \alpha_2 SQFT100 + \alpha_3 AGE + \alpha_4 AGE^2 + e$$

In [3]:
clear all

use https://www.stata.com/data/s4poe4/br2.dta

gen log_price = log(price)
gen age2 = age^2

reg log_price sqft age age2







      Source |       SS           df       MS      Number of obs   =     1,080
-------------+----------------------------------   F(3, 1076)      =    865.49
       Model |  209.891557         3  69.9638522   Prob > F        =    0.0000
    Residual |  86.9808098     1,076  .080837184   R-squared       =    0.7070
-------------+----------------------------------   Adj R-squared   =    0.7062
       Total |  296.872366     1,079  .275136577   Root MSE        =    .28432

------------------------------------------------------------------------------
   log_price |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
        sqft |   .0003876   8.69e-06    44.59   0.000     .0003706    .0004047
         age |  -.0175549   .0013565   -12.94   0.000    -.0202165   -.0148932
        age2 |   .0001734   .0000227     7.65   0.000     .0001289    .0002178
       _cons |   11.11959   .0274129   405.63

b. Interpret the estimate for $\alpha_2$

In this model, every $ \$1$increase in log price is predicted by an increase in 0.0003876 square feet.

c. Find and interpret estimates for $\frac{\partial \ln(PRICE)}{ \partial AGE}$ when $AGE = 5$ and $AGE = 20$

$$\frac{\partial \ln(PRICE)}{ \partial AGE} = \alpha_3 + 2\alpha_4 AGE = f(AGE)$$
$$ f(5) = -0.0175549 + 2 \times 0.0001734 \times 5 = -0.0158209$$
$$ f(20) = -0.0175549 + 2 \times 0.0001734 \times 20 = -0.0106189$$

d. Estimate $\frac{\partial PRICE}{\partial AGE}$ and $\frac{\partial PRICE}{\partial SQFT100}$ for a 20-year-old house with a living area of 2300 square feet.

$$PRICE(SQFT, AGE) = e^{\alpha_1 + \alpha_2 SQFT100} e^{\alpha_3 AGE+ \alpha_4AGE^2} $$
Find both partial differentials
$$ f(AGE, SQFT) = \frac{\partial PRICE(SQFT, AGE)}{\partial AGE} = e^{\alpha_1 + \alpha_2 SQFT100} e^{\alpha_3 AGE + \alpha_4AGE^2} (2 AGE \alpha_4 + \alpha_3) $$ 
$$ g(AGE, SQFT) =  \frac{\partial PRICE(SQFT, AGE)}{\partial SQFT100} =e^{\alpha_1 + \alpha_2 SQFT100} e^{\alpha_3 AGE+ \alpha_4AGE^2}\alpha_2 $$ 

Substitute in to find both elasticities
$$ f(20, 2300) = e^{11.11959\:+\:.0003876\cdot 2300}\:e^{-.0175549\cdot 20\:+\:.0001734\cdot 20^2}\:\left(2\:\cdot 20\:\cdot .0001734 -.0175549\right) = -1318.4500$$

$$ g(20, 2300) = e^{11.11959+0.0003876\cdot \:2300}e^{-0.0175549\cdot \:20+0.0001734\cdot \:20^2}\cdot 0.0003876 = 48.1246$$

Find the standard errors of your estimates in (e).

The form of f and g are not exactly tractable. Might have to use monte carlo simulation?

In [8]:
margins, dydx(sqft age)


Average marginal effects                        Number of obs     =      1,080
Model VCE    : OLS

Expression   : Linear prediction, predict()
dy/dx w.r.t. : sqft age

------------------------------------------------------------------------------
             |            Delta-method
             |      dy/dx   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
        sqft |   .0003876   8.69e-06    44.59   0.000     .0003706    .0004047
         age |  -.0175549   .0013565   -12.94   0.000    -.0202165   -.0148932
------------------------------------------------------------------------------



$$ SE(f) = \sqrt{ \text{var}(f)}$$

$$ SE(g) = \sqrt{\text{var}(g)}$$

Use the 2009 March CPS in Hansen's website (https://www.ssc.wisc.edu/~bhansen/econometrics/) and replicate equation 3.15 (the definition of experience is defined in footnote 9, pp. 17). Note that the regression is for the sub-sample of single (never married) Asian men. Rerun the regression in equation 3.14 using the sample defined here -- single (never married) Asian men -- and explain why the effect of education is smaller in equation 3.14 compared to 3.15.

Experience = age - education - 6


In [10]:
clear all

quiet use cps09mar.dta

gen wage = earnings / (hours*week) // wage is annual earnings/annual hours
gen log_wage = log(wage)
gen exper = age - education - 6
gen exper2 = exper^2

// regression for single Asian men
regress log_wage educ exper exper2 if marital == 7 & race == 4 & female == 0 










      Source |       SS           df       MS      Number of obs   =       268
-------------+----------------------------------   F(3, 264)       =     55.69
       Model |  52.5382617         3  17.5127539   Prob > F        =    0.0000
    Residual |  83.0162162       264  .314455365   R-squared       =    0.3876
-------------+----------------------------------   Adj R-squared   =    0.3806
       Total |  135.554478       267  .507694674   Root MSE        =    .56076

------------------------------------------------------------------------------
    log_wage |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
   education |   .1433165   .0116307    12.32   0.000     .1204157    .1662172
       exper |   .0355789   .0108576     3.28   0.001     .0142005    .0569574
      exper2 |  -.0007138   .0002957    -2.41   0.016     -.001296   -.0001315
       _cons |   .5753563   .1868299     3.

In [29]:
clear all

quiet use cps09mar.dta

drop if educ < 12

gen wage = earnings / (hours*week) // wage is annual earnings/annual hours
gen log_wage = log(wage)
gen exper = age - education - 6
gen exper2 = exper^2

gen fem_union = union == 1 & female == 1
gen male_union = union == 1 & female == 0
gen married_female =  marital == 1 | marital == 2 | marital == 3 & female == 1
gen married_male = marital == 1 | marital == 2 | marital == 3 & female == 0
gen formerly_married_female =  marital == 4 | marital == 5 & female == 1

gen hispanic = hisp == 1
gen black = race == 2
gen amerindian = race == 3
gen asian = race == 4
gen mixed = race > 5




(3,799 observations deleted)
















In [30]:
regress log_wage educ exper exper2 female fem_union male_union married_female married_male formerly_married_female hispanic black amerindian asian mixed


      Source |       SS           df       MS      Number of obs   =    46,943
-------------+----------------------------------   F(14, 46928)    =   1200.54
       Model |  5393.73773        14  385.266981   Prob > F        =    0.0000
    Residual |  15059.7097    46,928  .320910964   R-squared       =    0.2637
-------------+----------------------------------   Adj R-squared   =    0.2635
       Total |  20453.4475    46,942  .435717427   Root MSE        =    .56649

-----------------------------------------------------------------------------------------
               log_wage |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
------------------------+----------------------------------------------------------------
              education |   .1172723   .0011784    99.52   0.000     .1149626    .1195819
                  exper |   .0336282   .0008661    38.83   0.000     .0319306    .0353258
                 exper2 |  -.0005712    .000018   -31.80   0.000    -.0006

For regressors include years of education, potential work experience, experience-squared/100, and dummy variable indicators for the following: female, female union member, male union member, married female (marital code 1, 2, or 3), married male, formerly married female (marital code 4, 5, or 6), formerly married male, Hispanic, black, American Indian, Asian, and mixed race (race code 6 or higher).

In [12]:
regress log_wage educ if marital == 7 & race == 4 & female == 0 


      Source |       SS           df       MS      Number of obs   =       268
-------------+----------------------------------   F(1, 266)       =    144.23
       Model |  47.6587946         1  47.6587946   Prob > F        =    0.0000
    Residual |  87.8956834       266    .3304349   R-squared       =    0.3516
-------------+----------------------------------   Adj R-squared   =    0.3491
       Total |  135.554478       267  .507694674   Root MSE        =    .57483

------------------------------------------------------------------------------
    log_wage |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
   education |   .1411497   .0117531    12.01   0.000     .1180089    .1642906
       _cons |   .8584482   .1763195     4.87   0.000     .5112888    1.205608
------------------------------------------------------------------------------


The effect of education is smaller in the second equation because it is offset by a larger intercept. 

In [1]:
use https://www.stata.com/data/s4poe4/crime.dta

In [6]:
reg lcrmrte prbarr prbconv prbpris polpc wcon


      Source |       SS           df       MS      Number of obs   =       630
-------------+----------------------------------   F(5, 624)       =     59.65
       Model |  66.7397587         5  13.3479517   Prob > F        =    0.0000
    Residual |  139.640584       624  .223782987   R-squared       =    0.3234
-------------+----------------------------------   Adj R-squared   =    0.3180
       Total |  206.380342       629  .328108652   Root MSE        =    .47306

------------------------------------------------------------------------------
     lcrmrte |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
      prbarr |  -1.783391   .1164165   -15.32   0.000    -2.012006   -1.554775
     prbconv |  -.0963331   .0125767    -7.66   0.000    -.1210309   -.0716353
     prbpris |    .818684    .217159     3.77   0.000      .392233    1.245135
       polpc |   55.78244   8.121544     6.87   0.

In [13]:
reg lcrmrte prbarr polpc


      Source |       SS           df       MS      Number of obs   =       630
-------------+----------------------------------   F(2, 627)       =    100.16
       Model |  49.9716297         2  24.9858149   Prob > F        =    0.0000
    Residual |  156.408713       627  .249455682   R-squared       =    0.2421
-------------+----------------------------------   Adj R-squared   =    0.2397
       Total |  206.380342       629  .328108652   Root MSE        =    .49946

------------------------------------------------------------------------------
     lcrmrte |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
      prbarr |  -1.718258   .1215659   -14.13   0.000    -1.956983   -1.479532
       polpc |    25.8939   7.609941     3.40   0.001     10.94985    40.83796
       _cons |  -3.130721   .0410872   -76.20   0.000    -3.211406   -3.050036
--------------------------------------------------

In [7]:
reg lcrmrte prbarr prbconv prbpris polpc


      Source |       SS           df       MS      Number of obs   =       630
-------------+----------------------------------   F(4, 625)       =     74.40
       Model |    66.56898         4   16.642245   Prob > F        =    0.0000
    Residual |  139.811362       625   .22369818   R-squared       =    0.3226
-------------+----------------------------------   Adj R-squared   =    0.3182
       Total |  206.380342       629  .328108652   Root MSE        =    .47297

------------------------------------------------------------------------------
     lcrmrte |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
      prbarr |  -1.791017   .1160667   -15.43   0.000    -2.018945   -1.563089
     prbconv |  -.0965906   .0125709    -7.68   0.000    -.1212769   -.0719043
     prbpris |   .8114013   .2169578     3.74   0.000     .3853468    1.237456
       polpc |   55.54709   8.115535     6.84   0.

In [12]:
reg lcrmrte prbarr prbconv prbpris


      Source |       SS           df       MS      Number of obs   =       630
-------------+----------------------------------   F(3, 626)       =     77.88
       Model |  56.0892367         3  18.6964122   Prob > F        =    0.0000
    Residual |  150.291106       626  .240081638   R-squared       =    0.2718
-------------+----------------------------------   Adj R-squared   =    0.2683
       Total |  206.380342       629  .328108652   Root MSE        =    .48998

------------------------------------------------------------------------------
     lcrmrte |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
      prbarr |    -1.5484   .1144972   -13.52   0.000    -1.773245   -1.323555
     prbconv |  -.0571234   .0115722    -4.94   0.000    -.0798484   -.0343983
     prbpris |   .7755697   .2246969     3.45   0.001     .3343188    1.216821
       _cons |  -3.423979   .1065608   -32.13   0.

In [11]:
reg lcrmrte polpc


      Source |       SS           df       MS      Number of obs   =       630
-------------+----------------------------------   F(1, 628)       =      0.41
       Model |  .135241247         1  .135241247   Prob > F        =    0.5213
    Residual |  206.245101       628  .328415766   R-squared       =    0.0007
-------------+----------------------------------   Adj R-squared   =   -0.0009
       Total |  206.380342       629  .328108652   Root MSE        =    .57308

------------------------------------------------------------------------------
     lcrmrte |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
       polpc |  -5.361462   8.354889    -0.64   0.521    -21.76836    11.04544
       _cons |  -3.598948   .0278884  -129.05   0.000    -3.653714   -3.544183
------------------------------------------------------------------------------


In [8]:
reg lcrmrte prbarr


      Source |       SS           df       MS      Number of obs   =       630
-------------+----------------------------------   F(1, 628)       =    185.62
       Model |  47.0834389         1  47.0834389   Prob > F        =    0.0000
    Residual |  159.296903       628   .25365749   R-squared       =    0.2281
-------------+----------------------------------   Adj R-squared   =    0.2269
       Total |  206.380342       629  .328108652   Root MSE        =    .50364

------------------------------------------------------------------------------
     lcrmrte |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
      prbarr |   -1.59806   .1172959   -13.62   0.000      -1.8284   -1.367721
       _cons |  -3.118032   .0412608   -75.57   0.000    -3.199058   -3.037006
------------------------------------------------------------------------------


In [9]:
reg lcrmrte prbconv


      Source |       SS           df       MS      Number of obs   =       630
-------------+----------------------------------   F(1, 628)       =     23.38
       Model |  7.40663868         1  7.40663868   Prob > F        =    0.0000
    Residual |  198.973704       628  .316837108   R-squared       =    0.0359
-------------+----------------------------------   Adj R-squared   =    0.0344
       Total |  206.380342       629  .328108652   Root MSE        =    .56288

------------------------------------------------------------------------------
     lcrmrte |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
     prbconv |  -.0641963   .0132775    -4.83   0.000      -.09027   -.0381226
       _cons |  -3.565019    .024218  -147.21   0.000    -3.612577    -3.51746
------------------------------------------------------------------------------


In [10]:
reg lcrmrte prbpris


      Source |       SS           df       MS      Number of obs   =       630
-------------+----------------------------------   F(1, 628)       =     16.31
       Model |  5.22294784         1  5.22294784   Prob > F        =    0.0001
    Residual |  201.157395       628  .320314322   R-squared       =    0.0253
-------------+----------------------------------   Adj R-squared   =    0.0238
       Total |  206.380342       629  .328108652   Root MSE        =    .56596

------------------------------------------------------------------------------
     lcrmrte |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
     prbpris |   1.044458   .2586552     4.04   0.000      .536524    1.552392
       _cons |  -4.053661   .1123486   -36.08   0.000    -4.274286   -3.833037
------------------------------------------------------------------------------
